### ISSUE TITLE
Environment status is set to inactive

### ISSUE THREAD

**Issue by gergokee** [actions/runner#1001]
**Describe the bug**
If you have two jobs in same workflow, which uses same environment, even though the second job is running after the first one, the first jobs is rewriting the second job, causing the environment to be shown as "inactive".

**To Reproduce**
1.) create a workflow with two jobs with same environment and url which are running after eachother.
2.) run the workflow and see that the first job deactivates the second one, even though it should be the other way around.

**Expected behavior**
The job which was ran later should have the final word, and thus the environment would show up as active with the url.

**Comment by daveykane** [comment]
there is an option for deployments `auto_inactive` which can disable this behaviour. Can it be added to a jobs Environment properties along with the url?

**Comment by daveykane** [comment]
@yaananth can we revisit this one please? currently the behaviour is not as you describe.. There seems to be ONE active deployment per Environment Name rather than Environment URL which would be better?

**Comment by michaelst** [comment]
I am seeing the same thing as @daveykane

**Comment by CicerosPatience** [comment]
Yeah same here, we use a matrix to deploy to multiple k8s clusters in the same environment and it always reads `Inactive` a few minutes after the deploy job finishes successfully.

**Comment by error418** [comment]
This issue may be relate to #2120

**Comment by zburgermeiszter** [comment]
@yaananth 
> We always ensure there is ONE active deployment per URL.

Based on this, I ran some experiments.
If multiple apps are deployed against the same `environment`, as long as the **URL is unique** for each app in a (mono)repo both `deployment`s will show with a green tick.

A minor issue is only one of them will have the Active pill next to it, but at least not show a grey tick.

![image](https://github.com/actions/runner/assets/5986229/46a60b6a-ec84-47f8-8a44-8d852debac89)

**Comment by pegiadise** [comment]
Worked around it by adding a final job dependent on all the jobs and keeping the `environment` property only on that.

```yml
  deployment-status:
    needs: [deploy-job-1, deploy-job-2]
    runs-on: ubuntu-latest
    environment: production
    steps:
      - name: Deployment Completed
        run: echo "Production deployment completed successfully"
```
